docs/further_useful_features.rst

```diff
@@ -0,0 +1,438 @@
+previous contents.
+
+.. ipython:: python
+
+    from kartothek.io.eager import read_table
+
+    updated_df = read_table("a_unique_dataset_identifier", store_factory, table="table")
+    updated_df
+
+
+The way dataset updates work is that new partitions are added to a dataset
+as long as they have the same tables as the existing partitions. A `different`
+table **cannot** be introduced into an existing dataset with an update.
+
+To illustrate this point better, lets first create a dataset with two tables:
+
+.. ipython:: python
+
+    df2 = pd.DataFrame(
+        {
+            "G": "foo",
+            "H": pd.Categorical(["test", "train", "test", "train"]),
+            "I": np.array([9] * 4, dtype="int32"),
+            "J": pd.Series(3, index=list(range(4)), dtype="float32"),
+            "K": pd.Timestamp("20190604"),
+            "L": 2.,
+        }
+    )
+    df2
+
+    dm = store_dataframes_as_dataset(
+        store_factory,
+        "another_unique_dataset_identifier",
+        dfs = [
+            {
+                "data":
+                {
+                    "table1": df,
+                    "table2": df2
+                }
+            }
+        ]
+    )
+    dm.tables
+    dm.partitions
+
+
+Below is an example where we update the existing dataset ``another_unique_dataset_identifier``
+with new data for ``table1`` and ``table2``:
+
+.. ipython:: python
+
+    another_df2 = pd.DataFrame(
+        {
+            "G": "bar",
+            "H": pd.Categorical(["prod", "dev", "prod", "dev"]),
+            "I": np.array([12] * 4, dtype="int32"),
+            "J": pd.Series(4, index=list(range(4)), dtype="float32"),
+            "K": pd.Timestamp("20190614"),
+            "L": 10.,
+        }
+    )
+    another_df2
+
+    dm = update_dataset_from_dataframes(
+        {
+            "data":
+            {
+                "table1": another_df,
+                "table2": another_df2
+            }
+        },
+        store=store_factory,
+        dataset_uuid="another_unique_dataset_identifier"
+    )
+    dm.tables
+    dm.partitions
+
+
+Trying to update only a subset of tables throws a ``ValueError``:
+
+.. ipython::
+
+    @verbatim
+    In [45]: update_dataset_from_dataframes(
+       ....:        {
+       ....:           "data":
+       ....:           {
+       ....:              "table2": another_df2
+       ....:           }
+       ....:        },
+       ....:        store=store_factory,
+       ....:        dataset_uuid="another_unique_dataset_identifier"
+       ....:        )
+       ....:
+    ---------------------------------------------------------------------------
+    ValueError: Input partitions for update have different tables than dataset:
+    Input partition tables: {'table2'}
+    Tables of existing dataset: ['table1', 'table2']
+
+
+Adding data to existing datasets is not the only functionality achievable within an update
```

Comment:
Grammar:
Adding data to _an_ existing dataset~s~ is not the only functionality ...